### PR TITLE
docker_plugin: do not crash when plugin doesn't exist

### LINKED ISSUE #552: docker_plugin does not work in Ansible's check mode
##### SUMMARY

If you run a `docker_plugin` task with states "present" or "enable" (maybe "disable" too) with ansible in check_mode (flag `-C` in command line), the task fails because [it tries to query the docker API for information on the plugin when accessing the `result` property](https://github.com/ansible-collections/community.docker/blob/757b02cc154b2d545c82e62ac23469217f01e2a6/plugins/modules/docker_plugin.py#L353) even though it is in check mode and the plugin is bound to be absent - thus causing the failure. I'm not sure what the proper behavior should be, but I believe the check run should not fail here. 

The specific plugin I was trying to install is `grafana/loki-docker-driver:latest`, though I think that does not affect the issue.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

docker_plugin

##### ANSIBLE VERSION

```paste below
ansible 2.10.8
  config file = None
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.9.2 (default, Feb 28 2021, 17:03:44) [GCC 10.2.1 20210110]
```

##### COLLECTION VERSION

```paste below
# /root/.ansible/collections/ansible_collections
Collection       Version
---------------- -------
community.docker 3.3.2  
# /usr/lib/python3/dist-packages/ansible_collections
Collection       Version
---------------- -------
community.docker 1.2.2 
```

##### CONFIGURATION

`ansible-config dump --only-changed` output was empty

##### OS / ENVIRONMENT

This is happening to me when running a GitLab CI script on a "python" image and using roughly this sequence of commands for setup:

```bash
echo 'deb http://ppa.launchpad.net/ansible/ansible/ubuntu trusty main' > /etc/apt/sources.list.d/ansible.list
apt-key adv --keyserver keyserver.ubuntu.com --recv-keys # key omitted
apt-get update -y
apt-get install -y ansible
ansible-galaxy collection install community.docker
ansible-playbook deploy.yml -C # remaining arguments omitted
```

The target machine for the deploy runs an Ubuntu 22.04.

##### STEPS TO REPRODUCE

If I understood the module's code correctly, any attempt to install/enable a docker plugin via the `docker_plugin` module while running Ansible in check mode will fail. As an example, for me it failed using the following task:

```yaml
- name: Install Grafana Loki's driver for capturing logs
  community.docker.docker_plugin:
    plugin_name: grafana/loki-docker-driver:latest
    state: enable
    alias: loki
```

##### EXPECTED RESULTS

I'm not exactly sure what the `result` property should return in check mode (would `None` be acceptabe?) but the playbook as a whole should not fail since, in check mode, it assumes the plugin was properly installed.

##### ACTUAL RESULTS

This is the failure output only from the `docker_plugin` task. If more information is needed feel free to ask. I'm showing only this part because I'm afraid of inadvertedly sharing sensible information on our development environment. Also, this was enough for me to find the problem in the module's source code after a while.

```paste below
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_community.docker.docker_plugin_payload_ag90pjpr/ansible_community.docker.docker_plugin_payload.zip/ansible_collections/community/docker/plugins/module_utils/_api/api/client.py", line 259, in _raise_for_status
    response.raise_for_status()
  File "/usr/lib/python3/dist-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: http+docker://localhost/v1.41/plugins/loki/json
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/tmp/ansible_community.docker.docker_plugin_payload_ag90pjpr/ansible_community.docker.docker_plugin_payload.zip/ansible_collections/community/docker/plugins/modules/docker_plugin.py", line 375, in main
  File "/tmp/ansible_community.docker.docker_plugin_payload_ag90pjpr/ansible_community.docker.docker_plugin_payload.zip/ansible_collections/community/docker/plugins/modules/docker_plugin.py", line 353, in result
  File "/tmp/ansible_community.docker.docker_plugin_payload_ag90pjpr/ansible_community.docker.docker_plugin_payload.zip/ansible_collections/community/docker/plugins/module_utils/_api/api/client.py", line 532, in get_json
    return self._result(self._get(self._url(pathfmt, *args, versioned_api=True), **kwargs), json=True)
  File "/tmp/ansible_community.docker.docker_plugin_payload_ag90pjpr/ansible_community.docker.docker_plugin_payload.zip/ansible_collections/community/docker/plugins/module_utils/_api/api/client.py", line 266, in _result
    self._raise_for_status(response)
  File "/tmp/ansible_community.docker.docker_plugin_payload_ag90pjpr/ansible_community.docker.docker_plugin_payload.zip/ansible_collections/community/docker/plugins/module_utils/_api/api/client.py", line 261, in _raise_for_status
    raise_from(create_api_error_from_http_exception(e), e)
  File "/tmp/ansible_community.docker.docker_plugin_payload_ag90pjpr/ansible_community.docker.docker_plugin_payload.zip/ansible_collections/community/docker/plugins/module_utils/_api/errors.py", line 45, in create_api_error_from_http_exception
    raise_from(cls(e, response=response, explanation=explanation), e)
  File "<string>", line 3, in raise_from
ansible_collections.community.docker.plugins.module_utils._api.errors.NotFound: 404 Client Error for http+docker://localhost/v1.41/plugins/loki/json: Not Found ("plugin "loki" not found")
fatal: [swarm_manager]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "alias": "loki",
            "api_version": "auto",
            "ca_cert": null,
            "client_cert": null,
            "client_key": null,
            "debug": false,
            "docker_host": "unix://var/run/docker.sock",
            "enable_timeout": 0,
            "force_remove": false,
            "plugin_name": "grafana/loki-docker-driver:latest",
            "plugin_options": {},
            "ssl_version": null,
            "state": "enable",
            "timeout": 60,
            "tls": false,
            "tls_hostname": null,
            "use_ssh_client": false,
            "validate_certs": false
        }
    },
    "msg": "An unexpected docker error occurred: 404 Client Error for http+docker://localhost/v1.41/plugins/loki/json: Not Found (\"plugin \"loki\" not found\")"
}
```

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Fixes #552.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
docker_plugin

### CODE REVIEW

File: plugins/modules/docker_plugin.py [hunk 1]
```diff
@@ -346,11 +346,18 @@ def disable(self):
...
+                plugin_data = self.client.get_json('/plugins/{0}/json', self.preferred_name)
+            except NotFound:
+                # This can happen in check mode
+                pass
         result = {
             'actions': self.actions,
             'changed': self.changed,
             'diff': self.diff,
-            'plugin': self.client.get_json('/plugins/{0}/json', self.preferred_name) if self.parameters.state != 'absent' else {}
+            'plugin': plugin_data
```
Comment (briantist) [inline plugins/modules/docker_plugin.py:360]: optional nit/opinionated style preference
```suggestion
            'plugin': plugin_data,
```

**Review by briantist — APPROVED** [review]

File: plugins/modules/docker_plugin.py (hunk 1 above)
Comment (felixfontein) [reply plugins/modules/docker_plugin.py:360]: Damn, I completely forgot while being in a hurry... :) Thanks!

**Comment by felixfontein** [comment]
Re-triggering tests after 01429108d31248082bfa6f319b0cb3dfb0a012a2 was added to `main`.

**Comment by felixfontein** [comment]
@briantist thanks for reviewing this!

**Pull request merged**